### PR TITLE
Reduce dependence upon python-shogi

### PR DESCRIPTION
**Pull request by ddugovic**
- Log last move only if there is a last move
- Log last move after mate for variants
- Replace 50-move rule draw reason

### CODE REVIEW

File: lishogi-bot.py [hunk 1]
```diff
@@ -554,12 +560,12 @@ def tell_user_game_result(game, board):
...
     elif termination == engine_wrapper.Termination.DRAW:
-        if board.is_fifty_moves():
-            logger.info("Game drawn by 50-move rule.")
-        elif board.is_repetition():
-            logger.info("Game drawn by threefold repetition.")
-        else:
+        sdraw = game.state.get("sdraw")
+        gdraw = game.state.get("gdraw")
+        if sdraw and gdraw:
             logger.info("Game drawn by agreement.")
```
Comment (TheYoBots) [inline lishogi-bot.py:566]: Is it possible to agree to a draw in lishogi? I don't think so.

File: lishogi-bot.py [hunk 2]
```diff
@@ -554,12 +560,12 @@ def tell_user_game_result(game, board):
...
+        else:
+            logger.info("Game drawn by repetition (or impasse).")
```
Comment (TheYoBots) [inline lishogi-bot.py:568]: I don't think impasse is a draw either.

I know the goal in this PR is to reduce dependace on python-shogi, but over here having specific game endings for fourfold repetition and stalemate would be better using python-shogi:
> - Detects checkmates, stalemates.
> ```py
> >>> board.is_stalemate()
> False
>
> >>> board.is_game_over()
> True
> ```
>
> - Detects repetitions. Has a half move clock.
>```py
> >>> board.is_fourfold_repetition()
> False
>
> >>> board.move_number
> 8
> ```

Comment (TheYoBots) [reply lishogi-bot.py:568]: Or... if you don't want to use python-shogi, use this from engine_wrapper.py:
https://github.com/TheYoBots/Lishogi-Bot/blob/771a4eb4f30799c04fc5bdaa5ab356706f3d98ac/engine_wrapper.py#L39-L48

and write this part in the format:
```py
if termination == engine_wrapper.Termination.MATE:
        logger.info("Game won by checkmate.")
    elif termination == engine_wrapper.Termination.TIMEOUT:
        logger.info(f"{losing_name} forfeited on time.")
    elif termination == engine_wrapper.Termination.RESIGN:
        logger.info(f"{losing_name} resigned.")
    elif termination == engine_wrapper.Termination.ABORT:
        logger.info("Game aborted.")
# etc... just with TERMINATION.(gameEnding)...
```

File: lishogi-bot.py [hunk 3]
```diff
@@ -497,9 +501,12 @@ def fake_thinking(config, board, game):
...
 
 
-def print_move_number(board):
+def print_last_move(moves):
+    if moves:
+        moves = moves.split(" ")
+    move = moves[-1] if moves and len(moves) > 0 else None
     logger.info("")
-    logger.info(f"move: {len(board.move_stack) // 1 + 1}")
+    logger.info(f"Last move: {len(moves)}. {move}")
```
Comment (TheYoBots) [inline lishogi-bot.py:509]: Does move number get printed with this change? Because if it doesn't, I'd prefer it to be logged on every move.

Comment (ddugovic) [reply lishogi-bot.py:509]: This actually does print the move number `len(moves)`.

File: lishogi-bot.py (hunk 1 above)
Comment (ddugovic) [reply lishogi-bot.py:566]: In standard shogi, no; in chu shogi, yes (although it's possible Lishogi might not have implemented this).

File: lishogi-bot.py (hunk 3 above)
Comment (TheYoBots) [reply lishogi-bot.py:509]: Hmm... So for every move it prints `Last move: 2. 3c3d` (for example) right. Could it be changed to simply `move: 2. 3c3d` then?

```py
logger.info(f"move: {len(moves)}. {move}")
```

File: lishogi-bot.py (hunk 1 above)
Comment (TheYoBots) [reply lishogi-bot.py:566]: Right...Sorry. It is possible on lishogi to offer a draw only in Chushogi.

File: lishogi-bot.py (hunk 3 above)
Comment (ddugovic) [reply lishogi-bot.py:509]: I guess I can do those things...

This function prints the last move (by the opponent) and doesn't just print the move number, but I guess if the upstream repository does these things we might as well leave them be.

File: lishogi-bot.py (hunk 2 above)
Comment (ddugovic) [reply lishogi-bot.py:568]: Until python-shogi can fully support variants (and the Lishogi maintainer is in the middle of adding more variants!) depending on python-shogi may cause errors... I'll keep things simple as you recommend.

**Pull request merged**